src/Jira/Api/Client/CurlClient.php

```diff
@@ -122,4 +123,27 @@ public function sendRequest(
 
         return $data;
     }
+	
+	/**
+      * If necessary, replace curl file @ string with a CURLFile object (for PHP 5.5 and up)
+      *
+      * @param string $fileString The string in @-format as it is used on PHP 5.4 and older.
+      * @return \CURLFile|string
+      */
+     protected function getCurlValue($fileString)
+     {
+         $theCurlValue = $fileString;
+         $regex = "|^@(.*);filename=(.*)$|";
+         $matches = array();
+ 
+         if (function_exists('curl_file_create') && preg_match($regex, $fileString, $matches) === 1) {
+             $fileName = $matches[1];
```

Comment:
I believe I've made a mistake: In my original implementation I added an extra parameter to `\chobie\Jira\Api::createAttachment` to provide a name for the file to be uploaded. I did that in the format `'@' . $filename . ';filename=' . $postname` and I simply copy-pasted the function from my own implementation. I'll remove the clutter.